nima/http2/http2/src/main/java/io/helidon/nima/http2/ConnectionFlowControl.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.http2;
+
+import java.util.function.BiConsumer;
+
+import static java.lang.System.Logger.Level.DEBUG;
+
+/**
+ * HTTP/2 Flow control for connection.
+ */
+public class ConnectionFlowControl {
+
+    private static final System.Logger LOGGER_OUTBOUND = System.getLogger(FlowControl.class.getName() + ".ofc");
+
+    private final Type type;
+    private final BiConsumer<Integer, Http2WindowUpdate> windowUpdateWriter;
+    private final WindowSize.Inbound inboundConnectionWindowSize;
+    private final WindowSize.Outbound outboundConnectionWindowSize;
+    private int maxFrameSize = WindowSize.DEFAULT_MAX_FRAME_SIZE;
```

Comment:
If used from multiple threads, this will not be thread safe.

Reply:
thx fixed